tasks/build/download_node_builds.js

```diff
@@ -1,84 +1,134 @@
+  const platforms = grunt.config.get('platforms');
+  const downloadLimit = 3;
 
-        if (resp.statusCode !== 200) {
-          return cb(new Error(`${platform.nodeUrl} failed with a ${resp.statusCode} response`));
-        }
+  let shaSums = {};
+  const getShaSums = () => {
+    const nodeVersion = grunt.config.get('nodeVersion');
+    const shaSumsUri = `https://nodejs.org/dist/v${nodeVersion}/SHASUMS256.txt`;
 
-        return cb(null, resp);
+    return wreckGet(shaSumsUri).then(([resp, payload]) => {
+      if (resp.statusCode !== 200) {
+        throw new Error(`${shaSumsUri} failed with a ${resp.statusCode} response`);
+      }
+      payload.toString('utf8fix/7136 ').split('\n').forEach(line => {
+        const [sha, platform] = line.split('  ');
+        shaSums[platform] = sha;
       });
+      grunt.log.ok(`sha sums download complete`);
     });
+  };
 
-    // use an async iife to store promise for download
-    // then store platform in active downloads list
-    // which we will read from in the finish task
-    platform.downloadPromise = (async function () {
-      grunt.file.mkdir(downloadDir);
+  const checkShaSum = (platform) => {
+    const file = basename(platform.nodeUrl);
+    const path = resolve(platform.nodeDir, file);
+    const expected = {
+      hash: 'sha256',
+      expected: platform.win ? shaSums[basename(dirname(platform.nodeUrl)) + '/' + file] : shaSums[file]
+    };
+
+    return checkHashFromFile(path, expected).then(([passed, actual]) => {
+      if (passed) grunt.log.ok(`${platform.name} shasum verified`);
+      else grunt.log.error(`${platform.name} shasum check failed`);
+      return passed;
+    });
+  };
 
-      if (platform.win) {
-        await fromNode(cb => {
-          resp
-          .pipe(createWriteStream(resolve(downloadDir, 'node.exe')))
-          .on('error', cb)
-          .on('finish', cb);
-        });
-      } else {
-        await fromNode(cb => {
-          resp
-          .pipe(createGunzip())
-          .on('error', cb)
-          .pipe(new Extract({ path: downloadDir, strip: 1 }))
-          .on('error', cb)
-          .on('end', cb);
-        });
+  const getNodeBuild = (platform) => {
+    const finalDir = platform.nodeDir;
+    const downloadDir = `${finalDir}.temp`;
+    const file = basename(platform.nodeUrl);
+
+    return wreckGet(platform.nodeUrl)
+    .then(([resp, payload]) => {
+      if (resp.statusCode !== 200) {
+        throw new Error(`${platform.nodeUrl} failed with a ${resp.statusCode} response`);
       }
+      const bytes = parseInt(resp.headers['content-length'], 10) || 'unknown number of';
+      const mb = ((bytes / 1024) / 1024).toFixed(2);
+      grunt.log.ok(`downloading ${platform.name} - ${mb} mb`);
+      grunt.file.mkdir(downloadDir);
+      return payload;
+    })
+    .then(payload => {
+      return writeFileAsync(resolve(downloadDir, file), payload);
+    })
+    .then(() => {
+      return renameAsync(downloadDir, finalDir);
```

Comment:
Can you think of a reason why this renaming was happening in the old code? The only thing I can think of is that they didn't want to place a file in the "final" download dir until they were sure the download had completed successfully. But now that we're checking shasums, that seems unnecessary. Can you think of any other reason to keep this renaming step around?